ENIGMAsystem/SHELL/Platforms/General/POSIX/POSIXworkdir.cpp

```diff
@@ -6,18 +6,33 @@
 
 using std::string;
 
+namespace enigma {
+
+static inline string add_slash(const string& dir) {
+  if (dir.empty() || *dir.rbegin() != '/') return dir + '/';
+  return dir;
+}
+
+} // namespace enigma
+
 namespace enigma_user {
-  bool set_working_directory(string dname) {
-    if (!dname.empty()) {
-      while (*dname.rbegin() == '/') {
-        dname.erase(dname.size() - 1);
-      }
-    }
 
-    struct stat sb;
-    if (stat((char *)dname.c_str(), &sb) == 0 && S_ISDIR(sb.st_mode))
-      return (chdir((char *)dname.c_str()) == 0);
+bool set_working_directory(string dname) {
+  if (!dname.empty()) {
```

Comment:
This operation doesn't add anything; `chdir` will succeed regardless of duplicate or trailing slashes. Plus, you're just adding the slash below. If you want to canonicalize the output, you can use a call to `getcwd` after `chdir` completes.